### PR TITLE
Fix progress not covering nav/tab bar

### PR DESCRIPTION
**Pull request by michalsrutek**
This PR fixes cases when progress is shown, but due to the fact that the shadow view is inside viewcontroller's view, it doesn't cover navigation bar nor tab bar.

The fix adds the overlay to the viewcontroller's containing window if applicable.

First 2 images showcase the state before the fix, the latter ones show the same scenarios after the fix.

error1
error2
fixed1
fixed2

### CODE REVIEW

File: BT-Tracking/Interface/Data List/DataListVC.swift [hunk 1]
```diff
@@ -128,14 +127,14 @@ final class DataListVC: UIViewController, UITableViewDelegate {
...
-        present(controller, animated: true, completion: nil)
+        present(controller, animated: true)
     }
 
 }
 
 private extension DataListVC {
 
-    func sendReport() {
+    private func sendReport() {
```
Comment (augard) [inline BT-Tracking/Interface/Data List/DataListVC.swift:137]: Tady je private extension, to private je tu pridane zbytecne

File: BT-Tracking/Extensions/UIKit/Controller.swift [hunk 2]
```diff
@@ -27,8 +27,40 @@ extension UIViewController {
...
     func openURL(URL: URL) {
-        let controller = SFSafariViewController(url: URL)
-        present(controller, animated: true, completion: nil)
+        present(SFSafariViewController(url: URL), animated: true)
+    }
+
+    static let progressTag = 42
+
+    func showProgress() {
+        let viewToShowProgressOn = view.window ?? view
```
Comment (augard) [inline BT-Tracking/Extensions/UIKit/Controller.swift:36]: tady bude lepsi dat guard na to view

File: BT-Tracking/Extensions/UIKit/Controller.swift [hunk 3]
```diff
@@ -27,8 +27,40 @@ extension UIViewController {
...
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        activityIndicator.startAnimating()
+        overlay.addSubview(activityIndicator)
+
+        viewToShowProgressOn?.addSubview(overlay)
+
+        NSLayoutConstraint.activate([
+            activityIndicator.centerXAnchor.constraint(equalTo: overlay.centerXAnchor),
+            activityIndicator.centerYAnchor.constraint(equalTo: overlay.centerYAnchor),
+
```
Comment (augard) [inline BT-Tracking/Extensions/UIKit/Controller.swift:53]: Potom tady muzes pouzit to view a nemit force cast

File: BT-Tracking/Extensions/UIKit/Controller.swift [hunk 4]
```diff
@@ -27,8 +27,40 @@ extension UIViewController {
     }
 
     func openURL(URL: URL) {
-        let controller = SFSafariViewController(url: URL)
-        present(controller, animated: true, completion: nil)
+        present(SFSafariViewController(url: URL), animated: true)
+    }
+
+    static let progressTag = 42
```
Comment (augard) [inline BT-Tracking/Extensions/UIKit/Controller.swift:33]: klidne to udelaj private

File: BT-Tracking/Interface/Help/HelpVC.swift [hunk 5]
```diff
@@ -30,7 +30,7 @@ final class HelpVC: MarkdownController {
 
 private extension HelpVC {
 
-    func setupTabBar() {
+    private func setupTabBar() {
```
Comment (augard) [inline BT-Tracking/Interface/Help/HelpVC.swift:33]: + dalsi zbytecny private

File: BT-Tracking/Interface/Data List/DataListVC.swift (hunk 1 above)
Comment (michalsrutek) [reply BT-Tracking/Interface/Data List/DataListVC.swift:137]: Toho jsem si nevsiml. Opravim.

File: BT-Tracking/Interface/Help/HelpVC.swift (hunk 5 above)
Comment (michalsrutek) [reply BT-Tracking/Interface/Help/HelpVC.swift:33]: fixed... nejsem na private extensions zvykly :D

**Comment by michalsrutek** [comment]
@augard fixed and rebased

**Pull request merged**